packer/core.go

```diff
@@ -386,15 +399,30 @@ func (c *Core) init() error {
 	// We need to read the keys from both, then loop over all of them to figure
 	// out the appropriate interpolations.
 
-	allVariables := make(map[string]string)
+	repeatMap := make(map[string]string)
+	allKeys := make([]string, 0)
+
 	// load in template variables
 	for k, v := range c.Template.Variables {
-		allVariables[k] = v.Default
+		repeatMap[k] = v.Default
+		allKeys = append(allKeys, k)
 	}
 
 	// overwrite template variables with command-line-read variables
 	for k, v := range c.variables {
-		allVariables[k] = v
+		repeatMap[k] = v
+		allKeys = append(allKeys, k)
+	}
+
+	// sort map to force the following loop to be deterministic.
+	sort.Strings(allKeys)
+	type keyValue struct {
+		Key   string
+		Value string
+	}
+	sortedMap := make([]keyValue, len(repeatMap))
+	for _, k := range allKeys {
+		sortedMap = append(sortedMap, keyValue{k, repeatMap[k]})
```

Comment:
I might be too early but, don't forget to sort it 🙂 !

Reply:
It's getting sorted in 418.

Reply:
ah right ! I had a different implementation in mind !